cmake/FindAsio.cmake

```diff
@@ -0,0 +1,41 @@
+# Copyright (c)      2014 Thomas Heller
+# Copyright (c) 2007-2012 Hartmut Kaiser
+# Copyright (c) 2010-2011 Matt Anderson
+# Copyright (c) 2011      Bryce Lelbach
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+hpx_warn("finding asio")
+if(NOT TARGET Asio::asio)
+  find_path(
+    ASIO_INCLUDE_DIR asio.hpp
+    HINTS "${ASIO_ROOT}" ENV ASIO_ROOT "${HPX_ASIO_ROOT}"
+    PATH_SUFFIXES include
+  )
+
+  if(NOT ASIO_INCLUDE_DIR)
```

Comment:
Couldn't we silently fetch Asio automatically if we don't find any installed versions?

Reply:
I thought we concluded in the meeting that we don't want a silent fallback. I also prefer loudly stopping and telling users to explicitly opt-in to fetching Asio.

Reply:
My concern is that one of the arguments in favor of splitting our repository was to reduce the necessary configuration options for the users. This in fact increases the necessary options for a large number of users, however. However, in the end I am impartial to this change.